encoding/ssz/query/path.go

```diff
@@ -31,32 +36,89 @@ func ParsePath(rawPath string) ([]PathElement, error) {
 		if elem == "" {
 			return nil, errors.New("invalid path: consecutive dots or trailing dot")
 		}
+		snakeCasePath := toSnakeCase(elem)
 
-		fieldName := elem
-		var index *uint64
+		var pathElement PathElement
+		// Processing element string
+		processingField := snakeCasePath
 
-		// Check for index notation, e.g., "field[0]"
-		if strings.Contains(elem, "[") {
-			parts := strings.SplitN(elem, "[", 2)
-			if len(parts) != 2 {
-				return nil, fmt.Errorf("invalid index notation in path element %s", elem)
-			}
+		re := regexp.MustCompile(`^\s*len\s*\(\s*([^)]+?)\s*\)\s*$`)
+		matches := re.FindStringSubmatch(processingField)
+		if len(matches) == 2 {
+			pathElement.Length = true
+			// Extract the inner expression between len( and ) and continue parsing on that
+			processingField = matches[1]
+		}
 
-			fieldName = parts[0]
-			indexPart := strings.TrimSuffix(parts[1], "]")
-			if indexPart == "" {
-				return nil, errors.New("index cannot be empty")
-			}
+		// Default name is the full working string (may be updated below if it contains indices)
+		pathElement.Name = processingField
 
-			indexValue, err := strconv.ParseUint(indexPart, 10, 64)
+		if strings.Contains(processingField, "[") {
+			// Split into field and indices, e.g., "array[0][1]" -> name:"array", indices:{0,1}
+			pathElement.Name = extractFieldName(processingField)
+			indices, err := extractArrayIndices(processingField)
 			if err != nil {
-				return nil, fmt.Errorf("invalid index in path element %s: %w", elem, err)
+				return []PathElement{}, err
```

Comment:
```suggestion
				return nil, err
```

Same for the one at below.